libp2p/crypto/crypto.nim

```diff
@@ -76,8 +77,8 @@ type
   Signature* = object
     data*: seq[byte]
 
-  P2pKeyError* = object of CatchableError
-  P2pSigError* = object of CatchableError
+  P2pKeyError* = object of LibP2PError
```

Comment:
I think all the crypto stuff should be separate from libp2p? Is it generic enough to move to stew or some other place more appropriate @cheatfate?